src/nidigital/examples/burst_with_programmatic_configuration/nidigital_burst_with_programmatic_configuration.py

```diff
@@ -0,0 +1,184 @@
+        timing_filename = None if command == 'configure-time-set' else os.path.join(dir, 'Timing.digitiming')
+
+        session.load_specifications_levels_and_timing(spec_filename, levels_filename, timing_filename)
+
+        # Apply the settings from the levels and timing sheets we just loaded
+        session.apply_levels_and_timing(levels_filename if levels_filename is not None else '', timing_filename if timing_filename is not None else '')
+
+        # Configure voltage levels and terminal voltage through driver API
+        if command == 'configure-voltage':
+            session.channels[channels].configure_voltage_levels(voltage_config.vil, voltage_config.vih, voltage_config.vol, voltage_config.voh, voltage_config.vterm)
+            if voltage_config.termination_mode == 'High_Z':
+                session.channels[channels].termination_mode = nidigital.TerminationMode.HIGH_Z
+            elif voltage_config.termination_mode == 'Active_Load':
+                session.channels[channels].termination_mode = nidigital.TerminationMode.ACTIVE_LOAD
+                session.channels[channels].configure_active_load_levels(voltage_config.iol, voltage_config.ioh, voltage_config.vcom)
+            else:
+                session.channels[channels].termination_mode = nidigital.TerminationMode.VTERM
+
+        # Configure time set through driver API
+        if command == 'configure-time-set':
+            session.create_time_set(time_set_config.time_set_name)  # Must match time set name in pattern file
+            session.configure_time_set_period(time_set_config.time_set_name, time_set_config.period)
+            session.channels[channels].configure_time_set_drive_edges(time_set_config.time_set_name, convert_drive_format(time_set_config.drive_format),
+                                                                      time_set_config.drive_on, time_set_config.drive_data,
+                                                                      time_set_config.drive_return, time_set_config.drive_off)
+            session.channels[channels].configure_time_set_compare_edges_strobe(time_set_config.time_set_name, time_set_config.strobe_edge)
+
+        # Load the pattern file (.digipat) created using Digital Pattern Editor
+        pattern_filename = os.path.join(dir, 'Pattern.digipat')
+        session.load_pattern(pattern_filename)
+
+        # Burst pattern, blocks until the pattern is done bursting
+        session.burst_pattern(start_label='new_pattern')
+        print('Start bursting pattern')
+
+        # Disconnect all channels using programmable onboard switching
+        session.selected_function = nidigital.SelectedFunction.DISCONNECT
+    print('Done bursting pattern')
+
+
+def _main(argsv):
+    parser = argparse.ArgumentParser(description='Demonstrates how to create and configure an instrument session programmatically and to burst a pattern on the digital pattern instrument.', formatter_class=argparse.ArgumentDefaultsHelpFormatter)
```

Comment:
"Demonstrates how to create an instrument session, configure time set and voltage levels, and burst a pattern on the digital pattern instrument."